src/api/eth_ens.rs

```diff
@@ -0,0 +1,827 @@
+//! `Eth` namespace, ens
+
+use crate::{
+    api::Namespace,
+    contract::{Contract, Options},
+    signing::{namehash, NameHash},
+    types::{Address, TransactionId, U256},
+    Transport, Web3,
+};
+
+type ContractError = crate::contract::Error;
+type EthError = crate::ethabi::Error;
+
+use hex::ToHex;
+
+use idna::Config;
+
+const ENS_REGISTRY_ADDRESS: &str = "00000000000C2E074eC69A0dFb2997BA6C7d2e1e";
+
+const ADDR_INTERFACE_ID: &[u8; 4] = &[0x3b, 0x3b, 0x57, 0xde];
+const BLOCKCHAIN_ADDR_INTERFACE_ID: &[u8; 4] = &[0xf1, 0xcb, 0x7e, 0x06];
+const _NAME_INTERFACE_ID: &[u8; 4] = &[0x69, 0x1f, 0x34, 0x31];
+const _ABI_INTERFACE_ID: &[u8; 4] = &[0x22, 0x03, 0xab, 0x56];
+const PUBKEY_INTERFACE_ID: &[u8; 4] = &[0xc8, 0x69, 0x02, 0x33];
+const TEXT_INTERFACE_ID: &[u8; 4] = &[0x59, 0xd1, 0xd4, 0x3c];
+const CONTENTHASH_INTERFACE_ID: &[u8; 4] = &[0xbc, 0x1c, 0x58, 0xd1];
+
+/// `Eth` namespace, ens
+#[derive(Clone)]
+pub struct Ens<T: Transport> {
+    web3: Web3<T>,
+    registry: Registry<T>,
+    idna: Config,
+    transport: T,
+}
+
+impl<T: Transport> Namespace<T> for Ens<T> {
+    fn new(transport: T) -> Self
+    where
+        Self: Sized,
+    {
+        let web3 = Web3::new(transport.clone());
+
+        let registry = Registry::new(web3.eth());
+
+        let idna = Config::default()
+            .transitional_processing(false)
+            .use_std3_ascii_rules(true);
+
+        Self {
+            transport,
+            web3,
+            registry,
+            idna,
+        }
+    }
+
+    fn transport(&self) -> &T {
+        &self.transport
+    }
+}
+
+impl<T: Transport> Ens<T> {
+    //https://docs.ens.domains/contract-api-reference/name-processing#normalising-names
+    fn normalize_name(&self, domain: &str) -> Result<String, ContractError> {
+        self.idna
+            .to_ascii(domain)
+            .map_err(|_| ContractError::Abi(EthError::InvalidData))
+    }
+
+    /// Sets the resolver contract address of a name.
+    pub async fn set_resolver(
+        &self,
+        from: Address,
+        domain: &str,
+        address: Address,
+    ) -> Result<TransactionId, ContractError> {
+        let domain = self.normalize_name(domain)?;
+        let node = namehash(&domain);
+
+        self.registry.set_resolver(from, node, address).await
+    }
+
+    /// Returns the owner of a name.
+    pub async fn get_owner(&self, domain: &str) -> Result<Address, ContractError> {
```

Comment:
```suggestion
    pub async fn owner(&self, domain: &str) -> Result<Address, ContractError> {
```
I'd drop the verbs from method names when it's not necessary, since that's not a Rust convention.